CHANGELOG.md

```diff
@@ -2,5 +2,76 @@
 
 ## Release 2.5.0 (in development)
 
+### Breaking changes
+- Basic authentication has been deprecated in favour of OpenID Connect (OIDC)
+with Dex being the Identity provider of choice.
+This comes with the following:
+    - The Kubernetes API server authentication flag `--basic-auth` will be
+      removed and henceforth no longer be used in favour of `--oidc-issuer-url`.
+    - Fresh installations of MetalK8s will come with 2 Dex pods in a new
+      namespace `metalk8s-auth`
+    - Complete re-brand of the MetalK8s UI
+    - Authentication and access to MetalK8s UI and Grafana services will
+      henceforth implement OIDC based authentication
+
+- A new framework for persisting Cluster and Services Configurations (CSC) has
+been added to ensure configurations set by administrators are not lost during
+upgrade or downgrade.
+This comes with the following:
+    - New CSC's to hold configurations for:
+        - Dex configurations `metalk8s-dex-config`
+        - Grafana configurations `metalk8s-grafana-config`
+        - Prometheus configurations `metalk8s-prometheus-config`
+        - Alertmanager configurations `metalk8s-alertmanager-config`
+    - Procedures for changing and applying new configuration values for each
+      CSC has been documented.
```

Comment:
Let's rewrite it to make it more actionable (just a suggestion, feel free to reword):

```suggestion
User-provided configuration is now stored in ConfigMaps,
and MetalK8s tooling will honor the values provided when
deploying its services:
    - Dex uses `metalk8s-auth/metalk8s-dex-config`
    - Grafana uses `metalk8s-monitoring/metalk8s-grafana-config`
    - Prometheus uses `metalk8s-monitoring/metalk8s-prometheus-config`
    - Alertmanager uses `metalk8s-monitoring/metalk8s-alertmanager-config`
    
Documentation for changing and applying configuration values is
documented [here](...).
Note that any configuration applied on other Kubernetes objects
(e.g. a configuration Secret that Alertmanager uses, or the
Deployment of Grafana) will be lost on upgrade, and admins
should make sure to prepare the relevant ConfigMaps from their
existing configuration before upgrading to this version.
```